### PR TITLE
short names for matrix workflows

### PR DESCRIPTION
**Pull request by vlimant**
#### PR description:

while going through https://github.com/cms-sw/cms-bot/pull/1872 we discussed the painful long workflow names and subsequent long directory names.
this PR uses the mostly unused feature of defining a nickname for a workflow using the first string
`wf[num] = ['anickname', ['step1name', ...] ]`

the visual of `--show` and the `runall-report-step123-.log` are mostly unchanged

- [x] the PR will be complete with pulling some of the naming maintained in https://github.com/cms-sw/cms-bot/blob/master/comparisons/matrix_RE.txt 

but I'd like to check on opinions about this, and the interaction with cms-bot

#### PR validation:

configuring a couple of workflows with `runTheMatrix.py -l limited -j 0` went smoothly.

### CODE REVIEW

**Comment by smuzaffar** [comment]
please test

**Comment by vlimant** [comment]
please test

**Comment by vlimant** [comment]
please test

**Comment by vlimant** [comment]
@smuzaffar the comparison cannot function because the baseline has long names IMO. How can we test this ? on the sms-bot side maybe ?

**Comment by smuzaffar** [comment]
@vlimant , how about adding some login [here](https://github.com/cms-sw/cms-bot/blob/master/pr_testing/run-pr-comparisons#L106) to convert baseline long names to short names? e.g. something like
```
mv $WORKSPACE/data/$COMPARISON_RELEASE $WORKSPACE/data/${COMPARISON_RELEASE}-longnames
mkdir $WORKSPACE/data/$COMPARISON_RELEASE
#some logic to create symlinks $WORKSPACE/data/$COMPARISON_RELEASE/<wf>_<short_name> ../${COMPARISON_RELEASE}-longnames/<wf>_<longname> for each $WORKSPACE/data/$COMPARISON_RELEASE/<wf>_*
```

**Comment by smuzaffar** [comment]
by the way, the contents under `$WORKSPACE/data/$COMPARISON_RELEASE` should be identical to `/cvmfs/cms-ci.cern.ch/jenkins-artifacts/ib-baseline-tests/CMSSW_12_6_X_2022-11-22-2300/el8_amd64_gcc10/-GenuineIntel/matrix-results`

**Comment by vlimant** [comment]
while we fix the PR comparison mechanism. @cms-sw/pdmv-l2 @cms-sw/upgrade-l2 : does the change of directory name have an implication downstream ?

**Comment by srimanob** [comment]
> while we fix the PR comparison mechanism. @cms-sw/pdmv-l2 @cms-sw/upgrade-l2 : does the change of directory name have an implication downstream ?

Not from upgrade side.

**Comment by smuzaffar** [comment]
please test

**Comment by vlimant** [comment]
@smuzaffar can we restart this ?

**Comment by smuzaffar** [comment]
please test
Note there will be a lot of dqm differences as baseline has long names and this PR uses short names for workflows

**Comment by slava77** [comment]
I was looking at [the matrix outputs](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-353010/29197/runTheMatrix-results/)
Some of the automated short names for MC don't look quite clear because the text is the same: e.g.
- `5.1_TTbar`  vs `25.0_TTbar`: the first one is fastsim. Is there a way to add `FS` to fastsim MC?
- `138.[0-5]` workflows are already using non-descript short names. Please update to the year or  run; `Run3` suffix may be good enough
- nano workflows are too cryptic (e.g. `2500.601_mc126X`), having nano and perhaps also the MC origin would be useful

**Comment by smuzaffar** [comment]
please test with cms-sw/cms-bot#1897

**Comment by slava77** [comment]
it sounds like time to collect signatures

**Comment by vlimant** [comment]
@cms-sw/pdmv-l2 @cms-sw/upgrade-l2 : can you please take a look a sign if OK.

**Comment by sunilUIET** [comment]
+pdmv

File: Configuration/PyReleaseValidation/python/WorkFlowRunner.py [hunk 1]
```diff
@@ -207,7 +207,8 @@ def closeCmd(i,ID):
         logStat=''
         for i,s in enumerate(self.stat):
             logStat+='Step%d-%s '%(i,s)
-        self.report='%s_%s %s - time %s; exit: '%(self.wf.numId,self.wf.nameId,logStat,tottime)+' '.join(map(str,self.retStep))+'\n'
+        #self.report='%s_%s %s - time %s; exit: '%(self.wf.numId,self.wf.nameId,logStat,tottime)+' '.join(map(str,self.retStep))+'\n'
```
Comment (AdrianoDee) [inline Configuration/PyReleaseValidation/python/WorkFlowRunner.py:210]: Could you clean this up if not useful?

File: Configuration/PyReleaseValidation/python/MatrixReader.py [hunk 2]
```diff
@@ -308,8 +309,8 @@ def readMatrix(self, fileNameIn, useInput=None, refRel=None, fromScratch=None):
                 commands.append(cmd)
                 ranStepList.append(stepName)
                 stepIndex+=1
-                
-            self.workFlowSteps[(num,prefix)] = (num, name, commands, ranStepList)
+            #print('would use',name,'will use',name_for_workflow,'instead')
```
Comment (AdrianoDee) [inline Configuration/PyReleaseValidation/python/MatrixReader.py:312]: Also this, eventually.

**Comment by AdrianoDee** [comment]
@vlimant thanks for the PR (nice!). Really minor things, for the rest it's ready to go.

**Comment by vlimant** [comment]
sorry @sunilUIET for the signature reset

**Comment by vlimant** [comment]
please test with https://github.com/cms-sw/cms-bot/pull/1897

**Comment by AdrianoDee** [comment]
please test with https://github.com/cms-sw/cms-bot/pull/1897
(`CMSSW_13_0_X_2022-12-01-1100` is out)

**Comment by AdrianoDee** [comment]
+upgrade

**Comment by kskovpen** [comment]
+pdmv

**Comment by perrotta** [comment]
+1
- https://github.com/cms-sw/cms-bot/pull/1897 is now merged

**Comment by smuzaffar** [comment]
@vlimant , looks like after this change Relvals logs from IB page are not accessible. The IB page reads runall-report-step123-.log files to get the workflows names and this file still contains the long names [a]. Can you please update https://github.com/cms-sw/cmssw/blob/master/Configuration/PyReleaseValidation/python/WorkFlowRunner.py#L210 to also use short names?

[a] https://cmssdt.cern.ch/buildlogs/el8_amd64_gcc11/www/sat/13.0-sat-11/CMSSW_13_0_X_2022-12-03-1100/pyRelValMatrixLogs/run/runall-report-step123-.log
```
2500.311_NANOmc106Xul17v2+TTbarMINIAOD10.6_UL17v2+NANO_mc10.6ul17v2+HRV_NANO_mc Step0-PASSED Step1-PASSED Step2-PASSED  - time date Sat Dec  3 14:06:15 2022-date Sat Dec  3 14:06:12 2022; exit: 0 0 0
11634.914_TTbar_14TeV+2021_DDDDB+TTbar_14TeV_TuneCP5_2021_GenSim_DDDDB+Digi_DDDDB_2021+RecoNano_DDDDB_2021+HARVESTNano_DDDDB_2021+ALCA_DDDDB_2021 Step0-PASSED Step1-PASSED Step2-PASSED Step3-PASSED Step4-PASSED  - time date Sat Dec  3 14:05:49 2022-date Sat Dec  3 14:05:49 2022; exit: 0 0 0 0 0
11650.914_ZMM_14+2021_DDDDB+ZMM_14TeV_TuneCP5_2021_GenSim_DDDDB+Digi_DDDDB_2021+RecoNano_DDDDB_2021+HARVESTNano_DDDDB_2021+ALCA_DDDDB_2021 Step0-PASSED Step1-PASSED Step2-PASSED Step3-PASSED Step4-PASSED  - time date Sat Dec  3 14:05:49 2022-date Sat Dec  3 14:05:49 2022; exit: 0 0 0 0 0

```

**Comment by vlimant** [comment]
I had kept the step list in the final report so that one could see what step was failing directly. I overlooked that something was grabbing the directory names from that file (and honestly I was expecting that some script would). 
I am preparing a PR for this, however, would it be more natural to have the IB page do a listing of directories using the wf id from the report ?

**Comment by smuzaffar** [comment]
IBs page uses `runall-report-step123-.log` to get the results of each workflow e.g. how many steps are there and what are the exit codes for every step.

**Comment by vlimant** [comment]
I mean, can the parsing be done and mapping to the actual directory done by wf id, instead of removing the step list in #40235

**Comment by smuzaffar** [comment]
yes can be done but I would suggest to keep the `runall-report-step123-.log` and wf directory in sync. No idea how many other users might be using `runall-report-step123-.log`.  If you really want to have a full list of steps then I would suggest to generate a separate report file e.g. `runall-report-steplist123.log`

**Pull request merged**